### ISSUE TITLE
translate does not work for Exponential, CubicHermit, and ExponentialHermite splines

### ISSUE THREAD

**Issue by faymanns** [EPFL-Center-for-Imaging/splinebox#9]
The way `spline.translate(vector)` is currently implemented does not work for `Exponential`, `CubicHermite`, and `ExponentialHermite` spline curves.
The implementation simply adds the translation vector $\vec{v}$ to the coefficients:

$$f_{trans}(x) = \sum_k (c[k] + \vec{v}) \Phi(x - k)$$

$$= \sum_k c[k] \Phi(x - k) + \vec{v} \sum_k \Phi(x - k)$$

$$= f(x) + \vec{v} \sum_k \Phi(x - k)$$

If we want to satisfy the following definition of translation: $f_{trans}(x) = f(x) + \vec{v}$,
the basis function has to satisfy $\forall x \sum_k \Phi(x - k) = 1$.
I believe this this not true for the basis functions mentioned above.

We can potentially circumvent this problem by translating the knots instead and converting them back to coefficients after we added the translation vector.

**Comment by vuhlmann** [comment]
This is correct: translating the curve is only equal to a translation of the control points if the partition-of-unity condition is fulfilled. This is however the case for all basis functions you mention (Exponential, CubicHermite and ExponentialHermite), so if it doesn't work if means there is a bug somewhere.

The only trick in the Hermite case is that the translation only affects the 1st basis (as the second one interpolates derivatives, which are not affected by translation).

**Comment by vuhlmann** [comment]
As discussed, the bug is not in the code but in the fact that the exponential basis as implemented only satisfies the partition-of-unity condition for alpha=2*pi/M. This will affect both Exponential and ExponentialHermite. CubicHermite shouldn't have any issue unless there is indeed a bug.
